### PR TITLE
Add code coverage reporting

### PR DESCRIPTION
**Pull request by eNeRGy164**
* Adds code coverage to the build
* On local builds, report generator will generate an HTML report
  * Link to the report file is added to the output, if your terminal supports it, it's clickable
    image

* On CI builds, results will be added coveralls
  * https://coveralls.io/github/fluentassertions/fluentassertions
* After the first build with coverage is merged with master, subsequent PR's will have code coverage reported in the PR by coveralls

* Very small lay-out fixes (spaces, new-lines, etc.)

Didn't add it to the release notes as it's not modifying the library itself.

### CODE REVIEW

File: Build/Build.cs [hunk 1]
```diff
@@ -178,6 +194,8 @@ class Build : NukeBuild
                 .SetProject(Solution.Core.FluentAssertions)
                 .SetOutputDirectory(ArtifactsDirectory)
                 .SetConfiguration("Release")
+                .EnableNoLogo()
+                .EnableNoRestore()
```
Comment (dennisdoomen) [inline Build/Build.cs:198]: There was a reason I had to keep restoring the dependencies, but I can't remember why.

**Review by dennisdoomen — APPROVED** [review]

**Comment by eNeRGy164** [comment]
Hmm, I guess the target should be develop, not master

**Comment by eNeRGy164** [comment]
I did a bogus PR on my fork as a demo of a coveralls post: https://github.com/eNeRGy164/fluentassertions/pull/3#issuecomment-1043411729

**Comment by dennisdoomen** [comment]
> Hmm, I guess the target should be develop, not master

Correct.

**Pull request merged**